packages/screens/VideoPlayer/VideoPlayerHomeContent.tsx

```diff
@@ -0,0 +1,169 @@
+import { BrandText } from "../../components/BrandText";
+import { SVG } from "../../components/SVG";
+import { UploadVideoModal } from "../../components/videoPlayer/UploadVideoModal";
+import { VideoPlayerCard } from "../../components/videoPlayer/VideoPlayerCard";
+import {
+  combineFetchVideoPages,
+  useFetchVideos,
+} from "../../hooks/video/useFetchVideos";
+import { primaryColor } from "../../utils/style/colors";
+import { fontSemibold14, fontSemibold20 } from "../../utils/style/fonts";
+import { layout } from "../../utils/style/layout";
+import { VideoInfoWithMeta } from "../../utils/types/video";
+
+interface VideoPlayerProps {
+  req: GetVideoListRequest;
+  videoListForLibrary: VideoInfoWithMeta[];
+}
+
+export const VideoPlayerHomeContent: React.FC<VideoPlayerProps> = ({
+  req,
+  videoListForLibrary,
+}) => {
+  const [openUploadModal, setOpenUploadModal] = useState<boolean>(false);
+  const { data, isFetching, hasNextPage, fetchNextPage, isLoading } =
+    useFetchVideos(req);
+
+  const isLoadingValue = useSharedValue(false);
+  const isGoingUp = useSharedValue(false);
+  const [flatListContentOffsetY, setFlatListContentOffsetY] = useState(0);
+
+  const videos = useMemo(
+    () => (data ? combineFetchVideoPages(data.pages) : []),
+    [data]
+  );
+
+  const scrollHandler = useAnimatedScrollHandler({
+    onScroll: (event) => {
+      setFlatListContentOffsetY(event.contentOffset.y);
+      if (flatListContentOffsetY > event.contentOffset.y) {
+        isGoingUp.value = true;
+      } else if (flatListContentOffsetY < event.contentOffset.y) {
+        isGoingUp.value = false;
+      }
+      setFlatListContentOffsetY(event.contentOffset.y);
+    },
+  });
+  useEffect(() => {
+    if (isFetching || isLoading) {
+      isGoingUp.value = false;
+      isLoadingValue.value = true;
+    } else {
+      isLoadingValue.value = false;
+    }
+  }, [isFetching, isLoading, isGoingUp, isLoadingValue]);
+  const onEndReached = () => {
+    if (!isLoading && hasNextPage && !isFetching) {
+      fetchNextPage();
+    }
+  };
+
+  const styles = StyleSheet.create({
+    container: {
+      marginTop: layout.padding_x3,
+      width: "100%",
+    },
+    oneLine: {
+      flexDirection: "row",
+      justifyContent: "space-between",
+      alignItems: "center",
+    },
+    contentGroup: {
+      flexDirection: "row",
+      justifyContent: "center",
+      flexWrap: "wrap",
+      marginTop: layout.padding_x3,
+      gap: layout.padding_x2_5,
+      marginBottom: 40,
+    },
+    buttonGroup: {
+      flexDirection: "row",
+      alignItems: "center",
+      gap: layout.padding_x2,
+    },
+    buttonContainer: {
+      flexDirection: "row",
+      alignItems: "center",
+      paddingLeft: layout.padding_x1,
+      paddingRight: layout.padding_x1_5,
+      paddingVertical: layout.padding_x1,
+      backgroundColor: "#2B2B33",
+      borderRadius: layout.padding_x4,
+      gap: layout.padding_x1_5,
```

Comment:
```suggestion
      height: 32,
      backgroundColor: neutral30,
      borderRadius: 99,
      gap: layout.padding_x1,
```
(We want a rounded button, so the `borderRadius` can be very height to keep a perfectly rounded button without depending on its size IMO)